### PR TITLE
[AWSX-537] Support python version 3.10

### PR DESCRIPTION
**Pull request by ge0Aja**
### What does this PR do?

Support python version 3.10

### Motivation

Support request

### Additional Notes

This PR adds support for python version 3.10  which requires a higher version for `datadog-lambda` package.
The usage of the higher version for that package adds a new field `DD-API-KEY` in the header section of the of the events response forwarded by the function. It also removes the API key string from the `path` parameter.
 

### Types of changes

- [ ] Bug fix
- [ ] New feature
- [x] Breaking change
- [ ] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [x] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [x] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [x] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog
- [x] This PR passes the integration tests (ask a Datadog member to run the tests)
- [x] This PR passes the unit tests 
- [x] This PR passes the installation tests (ask a Datadog member to run the tests)

### CODE REVIEW

**Comment by ge0Aja** [comment]
had to use python versions as strings for triggering GH actions 
https://github.com/actions/runner/issues/1989

File: aws/logs_monitoring/tools/integration_tests/external_lambda/serverless.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 service: forwarder-tests-external-lambda
 provider:
   name: aws
-  runtime: python2.7
+  runtime: python3.11
```
Comment (tyrcho) [inline aws/logs_monitoring/tools/integration_tests/external_lambda/serverless.yml]: is this file really used ? I'm surprised in was on 2.X

File: aws/logs_monitoring/README.md [hunk 2]
```diff
@@ -138,6 +138,10 @@ If you can't install the Forwarder using the provided CloudFormation template, y
 
 If you encounter issues upgrading to the latest version, check the Troubleshooting section.
 
+### Upgrade an older version to +3.97.0
+
+Since version 3.97.0 the Lambda function has been updated to require **Python 3.11**. If upgrading an older forwarder installation to 3.97.0 or above, ensure the AWS Lambda function is configured to use Python 3.11
```
Comment (tyrcho) [inline aws/logs_monitoring/README.md]: it seems from most other files that we actually support 3.10 and 3.11 which is not really clear in this doc

File: aws/logs_monitoring/tools/Dockerfile_bundle [hunk 3]
```diff
@@ -19,4 +19,3 @@ RUN rm -rf ./botocore*
 # like `os.execl`, which cause security scans to fail for certain customers.
 # These files are not directly used by the Forwarder.
 RUN rm ./ddtrace/commands/ddtrace_run.py
-RUN rm ./decorator.py
```
Comment (tyrcho) [inline aws/logs_monitoring/tools/Dockerfile_bundle:22]: why this change ?

File: aws/logs_monitoring/tools/add_new_region.sh [hunk 4]
```diff
@@ -12,7 +12,7 @@ set -e
 
 OLD_REGION='us-east-1'
 
-PYTHON_VERSIONS_FOR_AWS_CLI=("python3.9")
+PYTHON_VERSIONS_FOR_AWS_CLI=("python3.11")
```
Comment (tyrcho) [inline aws/logs_monitoring/tools/add_new_region.sh]: do we support only one version here ? why an array then ?

File: aws/logs_monitoring/tools/Dockerfile_bundle (hunk 3 above)
Comment (ge0Aja) [reply aws/logs_monitoring/tools/Dockerfile_bundle:22]: file not found error was thrown during ITests, it seems it was added during a dependency version bump which created this file.
Now, it is not created, so rm is failing

File: aws/logs_monitoring/tools/integration_tests/external_lambda/serverless.yml (hunk 1 above)
Comment (ge0Aja) [reply aws/logs_monitoring/tools/integration_tests/external_lambda/serverless.yml]: caught that when I was doing the version changes, not sure if it is used.

File: aws/logs_monitoring/README.md (hunk 2 above)
Comment (ge0Aja) [reply aws/logs_monitoring/README.md]: we can update the readme to include both versions, I followed the convention in the readme file

File: aws/logs_monitoring/tools/add_new_region.sh (hunk 4 above)
Comment (ge0Aja) [reply aws/logs_monitoring/tools/add_new_region.sh]: it seems we support only the highest version, not sure why

File: .github/workflows/lint.yml [hunk 5]
```diff
@@ -11,7 +11,7 @@ jobs:
       - name: Setup Python
         uses: actions/setup-python@v3
         with:
-          python-version: 3.9
+          python-version: '3.10'
```
Comment (RaphaelAllier) [inline .github/workflows/lint.yml:14]: Why do we add quotes here?

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -8,7 +8,7 @@ jobs:
     strategy:
       max-parallel: 4
       matrix:
-        python-version: [3.8, 3.9]
+        python-version: ['3.9', '3.10']
```
Comment (RaphaelAllier) [inline .github/workflows/test.yml:11]: Same question, why do we use quotes now ?

File: aws/logs_monitoring/README.md [hunk 7]
```diff
@@ -138,6 +138,10 @@ If you can't install the Forwarder using the provided CloudFormation template, y
 
 If you encounter issues upgrading to the latest version, check the Troubleshooting section.
 
+### Upgrade an older version to +3.97.0
```
Comment (RaphaelAllier) [inline aws/logs_monitoring/README.md]: You need to update the cloudformation template we use to deploy the forwarder to ensure we use Python 3.10 as well

File: aws/logs_monitoring/setup.py [hunk 8]
```diff
@@ -12,12 +12,12 @@
...
     classifiers=[
-        "Programming Language :: Python :: 3.8",
         "Programming Language :: Python :: 3.9",
+        "Programming Language :: Python :: 3.10",
     ],
     keywords="datadog aws lambda layer",
-    python_requires=">=3.7, <3.10",
-    install_requires=["datadog-lambda==3.39.0", "requests-futures==1.0.0"],
+    python_requires=">=3.9, <3.11",
+    install_requires=["datadog-lambda==4.77.0", "requests-futures==1.0.0"],
```
Comment (RaphaelAllier) [inline aws/logs_monitoring/setup.py]: Where's that version change coming from ?

File: aws/logs_monitoring/tools/integration_tests/recorder/Dockerfile [hunk 9]
```diff
@@ -1,4 +1,4 @@
-FROM python:3.9
+FROM python:3.11
```
Comment (RaphaelAllier) [inline aws/logs_monitoring/tools/integration_tests/recorder/Dockerfile]: This still mentions 3.11 (PR title states 3.10)

File: aws/logs_monitoring/tools/publish_layers.sh [hunk 10]
```diff
@@ -13,7 +13,7 @@ set -e
 # Makes sure any subprocesses will be terminated with this process
 trap "pkill -P $$; exit 1;" INT
 
-PYTHON_VERSIONS_FOR_AWS_CLI=("python3.9")
+PYTHON_VERSIONS_FOR_AWS_CLI=("python3.11")
```
Comment (RaphaelAllier) [inline aws/logs_monitoring/tools/publish_layers.sh]: Still mentions 3.11 vs 3.10 in title of the PR

File: aws/logs_monitoring/setup.py (hunk 8 above)
Comment (ge0Aja) [reply aws/logs_monitoring/setup.py]: required dependency version update to support versions 3.10 -> 3.11

File: .github/workflows/test.yml (hunk 6 above)
Comment (ge0Aja) [reply .github/workflows/test.yml:11]: see https://github.com/actions/runner/issues/1989, same for the above

**Review by RaphaelAllier — APPROVED** [review]
